src/smk/tool/geomark/tool-geomark.js

```diff
@@ -151,33 +205,13 @@ include.module( 'tool-geomark', [
                     });
                 },
                 'create-geomark-from-file': function () {
-                    alert('Upload your file using the form in the new window. Once you have a Geomark URL, load it using "Load an Existing Geomark".');
+                    self.updateAndShowAlert('Upload your file using the form in the new window. Once you have a Geomark URL, add it to the map using "Load an Existing Geomark".');
```

Comment:
With the custom alert, we've now lost the desirable behavior where the app blocks until the user clicks ok and we then open the window.

Reply:
I've added dynamic handling of alerts so that we can conditionally open the "create geomark from file" window when closing an alert. I looked into a callback but it seems that would involve passing the callback function through the events, which I'm not sure we could do.

Reply:
Looks good.